src/compiler/checker.ts

```diff
@@ -15167,14 +15141,14 @@ namespace ts {
             return type.flags & TypeFlags.InstantiableNonPrimitive && maybeTypeOfKind(getBaseConstraintOfType(type) || emptyObjectType, TypeFlags.Nullable);
         }
 
-        function getConstraintForLocation(type: Type, node: Node): Type;
-        function getConstraintForLocation(type: Type | undefined, node: Node): Type | undefined;
-        function getConstraintForLocation(type: Type, node: Node): Type | undefined {
+        function getConstraintForLocation(type: Type, node?: Node): Type;
+        function getConstraintForLocation(type: Type | undefined, node?: Node): Type | undefined;
+        function getConstraintForLocation(type: Type, node?: Node): Type | undefined {

```

Comment:
Shouldn't this be `node: Node | undefined`, i.e. not an optional parameter? It seems pointless to make it optional since the function will never do anything if you don't pass an argument for it.

Also, when might `node` be `undefined`? Not clear to me what those changes are accomplishing.

Reply:
Didn't need to be `undefine`'d-able at all. I probably factored the place where it may have been undefinable so it wouldn't be while I was working on it - fix't.